src/Components/Comments.jsx

```diff
@@ -1,12 +1,42 @@
-import { useEffect, useState } from "react";
-import { getComments } from "../api";
+import { useContext, useEffect, useState } from "react";
+import { getComments, postComment } from "../api";
 import Loading from "./Loading";
+import { UserContext } from "../contexts/User";
 
 function Comments({ article_id }) {
-  const [commentInput, setCommentInput] = useState(""); //for post later//
+
+  const [badComment,setBadComment] = useState(false)
+  const [badPost, setBadPost] = useState(false)
   const [comment, setComment] = useState("");
+  const [success, setSuccess] = useState(true)
   const [comments, setComments] = useState([]);
   const [loading, setLoading] = useState(true)
+  const { user } = useContext(UserContext)
+  const [userData] = user
+
+  function handleChange(event){
+    setComment(event.target.value)
+  }
+
+  console.log(comments)
+
+  function handleSumbit(event){
+    event.preventDefault()
+    if(!comment){
+      setBadComment(true)
+    } else{
+      setBadComment(false)
+      return postComment(article_id,userData.username,comment)
+      .then(() => {
+        setBadPost(false)
+        setComment("")
+        setSuccess(true) //trigger useEffect - feedback appreicated
+      })
+      .catch((err) => {
+        setBadPost(true)
```

Comment:
You could update the comments state directly here like this:
` setComments((currComments) => [newComment, ...currComments])`

 I'd recommend having a look at these notes for reference: 
[https://notes.northcoders.com/courses/js-front-end/useEffect#cleanup-effects]